### PR TITLE
Organize job picker by role.

### PR DESCRIPTION
**Pull request by Juanaxu**
May this be the last time I have to move those job consts around.

### CODE REVIEW

File: client/src/app/party-select/party-select.component.html [hunk 1]
```diff
@@ -1,24 +1,58 @@
...
+	  </nz-avatar>
+  	</div>
+  	<div class="role-container">
+  	  <div class="role-name">Healer</div>
+  	  <nz-avatar class="party-member-icon"
+          *ngFor="let job of jobConsts.HEALER_JOBS;"
+	      [nzSize]="64" nzSrc="{{job.path}}" (click)="pickJob(job.name)">
+	  </nz-avatar>
+  	</div>
+  	<div class="role-container">
```
Comment (ackao) [inline client/src/app/party-select/party-select.component.html:36]: i think for the dps jobs there's no need to enforce roles. people run all sorts of interesting comps nowadays (and the dps sim will handle calculating the buff % for having different roles)

File: client/src/app/data/job.ts [hunk 2]
```diff
@@ -4,90 +4,134 @@ export interface Job {
...
+
+export const TANK_JOBS = [DARK_KNIGHT, GUNBREAKER, PALADIN, WARRIOR];
+
+export const HEALER_JOBS = [ASTROLOGIAN, SCHOLAR, WHITE_MAGE];
+
+export const MELEE_JOBS = [DRAGOON, MONK, NINJA, SAMURAI];
+
+export const RANGED_JOBS = [BARD, DANCER, MACHINIST];
+
+export const CASTER_JOBS = [BLACK_MAGE, RED_MAGE, SUMMONER];
```
Comment (ackao) [inline client/src/app/data/job.ts:137]: may be worth having an additional const `DPS_JOBS` (see other comment)

**Review by ackao — CHANGES_REQUESTED** [review]

File: client/src/app/data/job.ts (hunk 2 above)
Comment (Juanaxu) [reply client/src/app/data/job.ts:137]: Good idea, done.

File: client/src/app/party-select/party-select.component.html (hunk 1 above)
Comment (Juanaxu) [reply client/src/app/party-select/party-select.component.html:36]: Yeah, I don't have any role enforcement going on right now (you can make a party of 8 PLD if that's what you really want), this is more to make it easier to find what job you're looking for.

Comment (ackao) [reply client/src/app/party-select/party-select.component.html:36]: ack, i misunderstood what this code was supposed to do

**Review by ackao — APPROVED** [review]

**Pull request merged**